spec.bs

```diff
@@ -1502,7 +1524,70 @@ The <a spec="turtledove">validate and convert auction ad config</a> steps are
 modified to add the following steps just before the last step ("Return
 <var ignore>auctionConfig</var>"), renumbering the later step as appropriate:
 <div algorithm="protected-audience-validate-config-monkey-patch">
-7. If |config|[{{AuctionAdConfig/privateAggregationConfig}}] [=map/exists=]:
+31. If |config|["<code>{{AuctionAdConfig/auctionReportBuyerKeys}}</code>"]
+    [=map/exists=]:
+    1. Let |interestGroupBuyers| be |auctionConfig|'s <a spec="turtledove"
+        for="auction config">interest group buyers</a>.
+    1. If |interestGroupBuyers| is null, set |interestGroupBuyers| to a new
+        [=list=].
+    1. [=list/For each=] |index| of [=the exclusive range|the range=] 0 to
+        |config|["<code>{{AuctionAdConfig/auctionReportBuyerKeys}}</code>"]'s
+        [=list/size=], exclusive:
+        1. Let |key| be
+            |config|["<code>{{AuctionAdConfig/auctionReportBuyerKeys}}</code>"][|index|].
+        1. If |key|["{{PAHistogramContribution/bucket}}"] is not in the range
+            [0, 2<sup>128</sup>−1], [=exception/throw=] a {{TypeError}}.
+        1. If |index| is equal to or greater than |interestGroupBuyers|' [=list/
+            size=], [=iteration/break=].
+        1. Let |origin| be |interestGroupBuyers|[|index|].
+        1. [=map/Set=] |auctionConfig|'s [=auction config/auction report buyer
+            keys=][|origin|] to |key|.
+
+            Issue: Check behavior when an origin is repeated in
+                {{AuctionAdConfig/interestGroupBuyers}}.
+1. If |config|["<code>{{AuctionAdConfig/auctionReportBuyers}}</code>"] [=map/
+    exists=]:
+    1. [=map/iterate|For each=] |reportType| → |reportBuyerConfig| of
+        |config|["<code>{{AuctionAdConfig/auctionReportBuyers}}</code>"]:
+        1. If « "`interestGroupCount`", "`bidCount`",
+            "`totalGenerateBidLatency`", "`totalSignalsFetchLatency`" » does not
+            [=list/contain=] |reportType|, [=iteration/continue=].
+
+            Note: No error is thrown to allow forward compatibility if
+                additional report types are added later.
+
+            Issue: Should these strings be dash delimited?
+
+        1. If |reportBuyerConfig|["<code>{{AuctionReportBuyersConfig
+            /bucket}}</code>"] is not in the range [0, 2<sup>128</sup>−1],
+            [=exception/throw=] a {{TypeError}}.
+
+            Issue: Consider validating the case where the bucket used (after
+                summing) is too large. Currently, the implementation appears to
+                overflow. See <a
+                href="https://github.com/WICG/turtledove/issues/1040">
+                protected-audience/1040</a>.
+        1. [=map/Set=] |auctionConfig|'s [=auction config/auction report
+            buyers=][|reportType|] to |reportBuyerConfig|.
+1. If |config|["<code>{{AuctionAdConfig/requiredSellerCapabilities}}</code>"]
+    [=map/exists=]:
+    1. [=list/iterate|For each=] |capability| of
```

Comment:
[=list/For each=]. But as commented above, we can leave this to Protected Audience spec.

Reply:
Removed